doc/modules/mixture.rst

```diff
@@ -211,49 +211,78 @@ Cons
 
 .. _dpgmm:
 
-DPGMM: Infinite Gaussian mixtures
-=================================
+Dirichlet Process: Infinite Gaussian mixture
+============================================
+
+The :class:`DirichletGaussianMixture` class implements a variant of the
+Gaussian mixture model with a variable (but bounded) number of components using
+the Dirichlet Process.
 
-The :class:`DPGMM` object implements a variant of the Gaussian mixture
-model with a variable (but bounded) number of components using the
-Dirichlet Process.
 This class doesn't require the user to choose the number of
 components, and at the expense of extra computational time the user
 only needs to specify a loose upper bound on this number and a
 concentration parameter.
 
-.. |plot_gmm| image:: ../auto_examples/mixture/images/sphx_glr_plot_gmm_001.png
+
+The examples bellow compare Gaussian mixture models with fixed number of
+components, to the Gaussian mixture models with a Dirichlet process prior. Here,
+a classical Gaussian mixture is fitted with 5 components on a dataset composed
+of 2 clusters. We can see that the Dirichlet Gaussian mixture is able to limit
+itself to only 2 components whereas the Gaussian mixture fits the data with a
+fixed number of components that has to be set a priori by the user. In this case
+the user has select ``n_components=5`` which does not match the true generative
+distribution of this toy dataset. Note that with very little observations, the
+Dirichlet Gaussian Mixture can take a conservative stand, and fit only one
+component.
+
+.. figure:: ../auto_examples/mixture/images/sphx_glr_plot_gmm_001.png
    :target: ../auto_examples/mixture/plot_gmm.html
-   :scale: 48%
+   :align: center
+   :scale: 70%
 
-.. |plot_gmm_sin| image:: ../auto_examples/mixture/images/sphx_glr_plot_gmm_sin_001.png
-   :target: ../auto_examples/mixture/plot_gmm_sin.html
-   :scale: 48%
 
-.. centered:: |plot_gmm| |plot_gmm_sin|
+On the following example, we are fitting a dataset not well-depicted by a Gaussian
+mixture. Adjusting the ``weight_concentration_prior``, parameter of the DPGMM
+controls the number of components used to fit this data.
 
+.. figure:: ../auto_examples/mixture/images/sphx_glr_plot_gmm_sin_001.png
+   :target: ../auto_examples/mixture/plot_gmm_sin.html
+   :align: center
+   :scale: 65%
+
+The next figure presents the resulting clusters computed by the Gaussian mixture
+with a Dirichlet process prior for different values of
+``weight_concentration_prior``. The ``weight_concentration_prior`` is related to
+the number of clusters obtained.  As in Variational Bayesian Gaussian Mixture
+(for the ``weight_concentration_prior`` parameter), smaller values of
```

Comment:
I think sample method is useful to have a generative model that can be used for a classifier eg to model p(x|y)
It does more than scipy






On Tue, Sep 6, 2016 at 4:33 PM +0200, "Olivier Grisel" <notifications@github.com> wrote:












In doc/modules/mixture.rst:
>  
> +.. figure:: ../auto_examples/mixture/images/sphx_glr_plot_gmm_sin_001.png
> +   :target: ../auto_examples/mixture/plot_gmm_sin.html
> +   :align: center
> +   :scale: 65%
> +
> +The next figure presents the resulting clusters computed by the Gaussian mixture
> +with a Dirichlet process prior for different values of
> +``weight_concentration_prior``. The ``weight_concentration_prior`` is related to
> +the number of clusters obtained.  As in Variational Bayesian Gaussian Mixture
> +(for the ``weight_concentration_prior`` parameter), smaller values of



Please remove "(for the weight_concentration_prior parameter)" which is redundant now that the two parameters have the same name.



—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.